src/Tizen.Wearable.CircularUI.Forms.Renderer/Shell/NavigationDrawer.cs

```diff
@@ -0,0 +1,322 @@
+            _contentBox.PackEnd(content);
+
+            if (content is NavigationView nv)
+            {
+                nv.Dragged += (s, e) =>
+                {
+                    if (e.State == DraggedState.EdgeTop)
+                    {
+                        Close();
+                    }
+                };
+            }
+        }
+
+        public async void Open(uint length = 300)
+        {
+            var toMove = _drawer.Geometry;
+            toMove.Y = 0;
+
+            await RunMoveAnimation(_drawer, toMove, length);
+
+            if (!_isOpen)
+            {
+                _isOpen = true;
+                Toggled?.Invoke(this, EventArgs.Empty);
+            }
+        }
+
+        public async void Close(uint length = 300)
+        {
+            var toMove = _drawer.Geometry;
+            toMove.Y = Geometry.Height - _iconHeight;
+
+            await RunMoveAnimation(_drawer, toMove, length);
+
+            if (_isOpen)
+            {
+                _isOpen = false;
+                Toggled?.Invoke(this, EventArgs.Empty);
+            }
+        }
+
+        public async Task<bool> ShowAsync(EWidget target, Easing easing = null, uint length = 300, CancellationToken cancelltaionToken = default(CancellationToken))
+        {
+            var tcs = new TaskCompletionSource<bool>();
+
+            await Task.Delay(1000);
+
+            if (cancelltaionToken.IsCancellationRequested)
+            {
+                cancelltaionToken.ThrowIfCancellationRequested();
+            }
+
+            target.Show();
+            var opacity = target.Opacity;
+
+            new Animation((progress) =>
+            {
+                target.Opacity = opacity + (int)((255 - opacity) * progress);
+
+            }).Commit(this, "FadeIn", length: length, finished: (p, e) =>
+            {
+                target.Opacity = 255;
+                tcs.SetResult(true);
+            });
+
+            return await tcs.Task;
+        }
+
+        public async Task<bool> HideAsync(EWidget target, Easing easing = null, uint length = 300)
+        {
+            var tcs = new TaskCompletionSource<bool>();
+
+            var opacity = target.Opacity;
+            new Animation((progress) =>
+            {
+                target.Opacity = opacity - (int)(progress * opacity);
+
+            }).Commit(this, "FadeOut", length: length, finished: (p, e) =>
+            {
+                target.Opacity = 0;
+                target.Hide();
+                tcs.SetResult(true);
+            });
+
+            return await tcs.Task;
+        }
+
+        protected override IntPtr CreateHandle(EvasObject parent)
+        {
+            _mainLayout = new Box(parent);
+            return _mainLayout.Handle;
+        }
+
+        void Initialize()
+        {
+            AlignmentX = -1;
+            AlignmentY = -1;
+            WeightX = 1;
+            WeightY = 1;
+
+            _mainLayout.RepeatEvents = true;
+
+            _drawer = new Box(this);
```

Comment:
Nobody contains `_drawer`. You should build a complete view tree. currently _drawer has no parent

Reply:
Fixed.

Reply:
`_drawer` has no parent again